{
  try {
    exitReasonQ.put(reason);
  }
 catch (  InterruptedException e) {
    e.printStackTrace();
  }
}
